src/Mvc/Mvc.Core/src/StatusCodeResult.cs

```diff
@@ -39,12 +42,28 @@ public override void ExecuteResult(ActionContext context)
                 throw new ArgumentNullException(nameof(context));
             }
 
-            var factory = context.HttpContext.RequestServices.GetRequiredService<ILoggerFactory>();
+            Execute(context.HttpContext);
+        }
+
+        /// <summary>
+        /// Sets the status code on the HTTP response.
+        /// </summary>
+        /// <param name="httpContext">The <see cref="HttpContext"/> for the current request.</param>
+        /// <returns>A task that represents the asynchronous execute operation.</returns>
+        public ValueTask WriteResponseAsync(HttpContext httpContext)
```

Comment:
We would explicitly implement these.

Reply:
How iffy is it to call `ExecuteResult` instead with a cooked up `ActionContext`? Depending on how you implement it, it would mean you wouldn't have to change MVC's result types.

Reply:
> How iffy is it to call ExecuteResult instead with a cooked up ActionContext?

That feels very fragile. I could see it leading to weird runtime errors when edge cases are hit.

Also, do we want to be using `IActionResultExecutor<TResult>` implementations? This PR doesn't currently for `JsonResult.WriteResponseAsync(HttpContext)` unlike `JsonResult.ExecuteAsync(ActionContext)`. Using them would give us extra functionality and extensibility but would add overhead.

Reply:
I think it's doable with endpoint routing. All of the information is in the HttpContext that's required to make a real `ActionContext`

Reply:
Producing an `ActionDescriptor` would be the issue. It's not really something I would expect or want to stub. We could choose to make it null but that does increase the risk of seeing a runtime error.

Reply:
Are you suggesting some shared code in Mvc.Core that creates an ActionContext from an HttpContext for the various IActionResult implementations to use in their `IActionResult.ExecuteResult(HttpContext context)` implementations in order to call `IActionResult.ExecuteResult(ActionContext context)`?

We still cannot have the MapAction implementation itself create the ActionContext without ruining the layering. This means that all the IActionResults that we want to support with MapAction will need to be updated to implement a new interface anyway.

If everything can be rewritten in terms of HttpContext, why not just have the implementations of `IActionResult.ExecuteResult(ActionContext context)` call `IActionResult.ExecuteResult(HttpContext context)` instead? Seems easier than having to create fake ActionContexts and ActionDescriptors which I doubt would fully work anyway. Would we even try to support something trying to add a FilterDescriptor?

Reply:
Leave it as is.